motion_control.c

```diff
@@ -247,7 +247,9 @@ void mc_arc (float *target, plan_line_data_t *pl_data, float *position, float *o
     if(labs(turns) > 1) {
 
         uint32_t n_turns = labs(turns) - 1;
-        float arc_travel = 2.0f * M_PI * n_turns + angular_travel;
+        float arc_travel = 2.0f * M_PI * n_turns - angular_travel;
+		if (turns > 0)
+			arc_travel = 2.0f * M_PI * n_turns + angular_travel;
```

Comment:
IMO better to use the ternary operator, likely results in less code emitted (the STM32F1xx driver is tight on flash):

` float arc_travel = 2.0f * M_PI * (float)n_turns + (turns > 0 ? angular_travel : -angular_travel);`

Reply:
> At least for Teensy 4.1 using Arduino IDE2 that does not matter

Ok, then the compiler optimizes away the duplicated parts of the code - not bad. Anyway, IMO by using ternary operator it makes the code simpler and a bit easier to read.